### ISSUE TITLE
Writing to an invalid stream creates an infinite loop of errors

### ISSUE THREAD

**Issue by onnoschutze** [reactphp/stream#2]
In the context of React v0.4.1 in the Ratchet Websocket library (0.3.1) I encountered the following situation. When the buffer code somehow tries to write to an invalid stream (Buffer.php L83) it will emit the error "tried to write to invalid stream". 

Buffer.php
```php
public function handleWrite()
{
   if (!is_resource($this->stream)) {
       $this->emit('error', array(new \RuntimeException('Tried to write to invalid stream.'), $this));
       return;
   }
   ...
```

The stream will receive this error & emits the error also. Somehow this triggers writing to the buffer again, which then emits the "tried to write to invalid stream" error again, thus creating an infinite loop.

Stream.php
```php
public function __construct($stream, LoopInterface $loop)
{
   ...

   $this->buffer->on('error', function ($error) {
        $this->emit('error', array($error, $this));
        $this->close();
   });

   ...
}
```

Removing the " $this->emit('error', array($error, $this));" from the Stream __construct fixes the infinite loop, but it seems to be that not the real problem is fixed in this way. I have constructed a piece of code that enables me to reproduce this problem (via websockets). I will try to create a unit test that fails for this piece of code. But I hope this report helps as a start.

**Comment by lysenkobv** [comment]
Hi, take a look on #25. Same issue.

**Comment by clue** [comment]
We believe this has been fixed as part of #25 and #40 :+1: A release will be tagged soon, please report back if this problem persists!
